dali/test/python/test_operator_video_reader_resize.py

```diff
@@ -0,0 +1,155 @@
+import os
+from functools import partial
+
+import numpy as np
+import nvidia.dali as dali
+import nvidia.dali.ops as ops
+import nvidia.dali.types as types
+from nvidia.dali.pipeline import Pipeline
+from PIL import Image as Image
+
+
+video_directory = '/tmp/labelled_videos/'
+video_directory_multiple_resolutions = '/tmp/video_resolution/'
+
+pipeline_params = {
+    'num_threads': 8,
+    'device_id': 0,
+    'seed': 0
+}
+
+
+video_reader_params = [{
+    'device': 'gpu',
+    'file_root': video_directory,
+    'sequence_length': 32,
+    'random_shuffle': False
+}, {
+    'device': 'gpu',
+    'file_root': video_directory_multiple_resolutions,
+    'sequence_length': 4,
+    'random_shuffle': False
+}]
+
+
+resize_params = [{
+    'resize_x': 300,
+    'resize_y': 200,
+    'interp_type': types.DALIInterpType.INTERP_CUBIC,
+    'minibatch_size': 8
+}, {
+    'resize_x': 300,
+    'interp_type': types.DALIInterpType.INTERP_CUBIC,
+    'minibatch_size': 8
+}, {
+    'resize_x': 300,
+    'resize_y': 200,
+    'interp_type': types.DALIInterpType.INTERP_LANCZOS3,
+    'minibatch_size': 8
+}, {
+    'resize_shorter': 300,
+    'interp_type': types.DALIInterpType.INTERP_CUBIC,
+    'minibatch_size': 8
+}, {
+    'resize_longer': 500,
+    'interp_type': types.DALIInterpType.INTERP_CUBIC,
+    'minibatch_size': 8
+}, {
+    'resize_x': 300,
+    'resize_y': 200,
+    'min_filter': types.DALIInterpType.INTERP_CUBIC,
+    'mag_filter': types.DALIInterpType.INTERP_TRIANGULAR,
+    'minibatch_size': 8
+}, {
+    'resize_x': 300,
+    'resize_y': 200,
+    'interp_type': types.DALIInterpType.INTERP_CUBIC,
+    'minibatch_size': 4
+}]
+
+
+def video_reader_pipeline_base(
+        video_reader, batch_size, video_reader_params, resize_params={}):
+    pipeline = dali.pipeline.Pipeline(
+        batch_size=batch_size, **pipeline_params)
+
+    with pipeline:
+        outputs = video_reader(
+            **video_reader_params, **resize_params)
+        if type(outputs) == list:
+            outputs = outputs[0]
+        pipeline.set_outputs(outputs)
+    pipeline.build()
+
+    return pipeline
+
+
+def video_reader_resize_pipeline(batch_size, video_reader_params, resize_params):
+    return video_reader_pipeline_base(
+        dali.fn.video_reader_resize, batch_size, video_reader_params, resize_params)
+
+
+def video_reader_pipeline(batch_size, video_reader_params):
+    return video_reader_pipeline_base(
+        dali.fn.video_reader, batch_size, video_reader_params)
+
+
+def ground_truth_pipeline(batch_size, video_reader_params, resize_params):
+    def get_next_frame():
+        pipeline = video_reader_pipeline(batch_size, video_reader_params)
+
+        pipe_out = pipeline.run()
+        sequences_out = pipe_out[0].as_cpu().as_array()
+        for sample in range(batch_size):
+            yield [sequences_out[sample]]
+
+    gt_pipeline = dali.pipeline.Pipeline(
```

Comment:
Maybe you can use ElementExtract to extract all frames from a sequence as a separate batches and call resize on it instead of going through ExternalSource?

Reply:
Maybe I'm mistaken, but I think that writing a pipeline with `ElementExtract` that would be robust to changes of `batch_size` and `sequence_length` throughout the tests might be somewhat involved. You need to adjust number of outputs and calls to `Resize` depending on `sequence_length`. If you feel strongly about it, I can do it, but maybe we can leave it as it is?

Reply:
I think it would depend only on the `sequence_length`, and you can do it in the loop. So you can compare two pipelines:
VideoReader+seq_len*ElementExtract+seq_len*Resize vs VideoesizeReader+seq_len*ElementExtract. But it is just an idea. Let us wait for another opinion.

Reply:
Done